core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -1103,8 +1103,8 @@ void parallel_reduce(const size_t& policy,
                                      , typename ValueTraits::pointer_type
                                      >::type value_type ;
 
-  static_assert(Impl::FunctorAnalysis<Impl::FunctorPatternInterface::REDUCE,RangePolicy<>,FunctorType>::
-                 has_final_member_function,"Calling parallel_reduce without either return value or final function.");
+ // static_assert(Impl::FunctorAnalysis<Impl::FunctorPatternInterface::REDUCE,RangePolicy<>,FunctorType>::
```

Comment:
Y'all gotta _commit_ -- either delete it or leave it, don't just commit it out :-)

Reply:
We actually need the equivalent of this check, it stopped working due to the policy (default execution space) being an incomplete type, but in theory we shouldn't need the policy to check what we're checking here. Options: fix Kokkos headers so execution spaces show up first, or, come up with an alternative to `FunctorAnalysis` to check for a final method.

Reply:
Current preferred solution: add `has_final_method<FunctorType>`, use it here. Even further, I think we're considering getting rid of final methods, but for now this is needed.